hackathon_site/review/tests.py

```diff
@@ -1,3 +1,356 @@
+            ).all()
+        )
+
+        self.assertEqual(len(mail.outbox), 3)
+        self.assertRedirects(response, self.view)
+        self.assertEqual(quantity_before, quantity_after + self.form_data["quantity"])
+
+    def test_sends_quantity_number_of_waitlisted_emails(self):
+        self._login()
+        self._create_teams_and_reviews_for_mail_tests()
+
+        quantity_before = len(
+            Review.objects.filter(
+                decision_sent_date__isnull=True, status="Waitlisted"
+            ).all()
+        )
+
+        # Send only two waitlisted email
+        self.form_data["status"] = "Waitlisted"
+        self.form_data["quantity"] = 2
+        response = self.client.post(self.view, data=self.form_data)
+
+        quantity_after = len(
+            Review.objects.filter(
+                decision_sent_date__isnull=True, status="Waitlisted"
+            ).all()
+        )
+
+        self.assertEqual(len(mail.outbox), 2)
+        self.assertRedirects(response, self.view)
+        self.assertEqual(quantity_before, quantity_after + self.form_data["quantity"])
+
+    def test_sends_quantity_number_of_rejected_emails(self):
+        self._login()
+        self._create_teams_and_reviews_for_mail_tests()
+
+        quantity_before = len(
+            Review.objects.filter(
+                decision_sent_date__isnull=True, status="Rejected"
+            ).all()
+        )
+
+        self.form_data["status"] = "Rejected"
+        self.form_data["quantity"] = 3  # Send 3 rejection emails
+
+        response = self.client.post(self.view, data=self.form_data)
+
+        quantity_after = len(
+            Review.objects.filter(
+                decision_sent_date__isnull=True, status="Rejected"
+            ).all()
+        )
+
+        self.assertEqual(len(mail.outbox), 3)
+        self.assertRedirects(response, self.view)
+        self.assertEqual(quantity_before, quantity_after + self.form_data["quantity"])
+
+    def test_send_all_accepted_within_current_date_range(self):
+        self._login()
+        self._create_teams_and_reviews_for_mail_tests(date_offset=5)
+
+        quantity_before = len(
+            Review.objects.filter(
+                decision_sent_date__isnull=True, status="Accepted"
+            ).all()
+        )
+
+        # Send 10 acceptance emails. There's only 8 accepted people, and 1 older. So should
+        # only send 7 emails
+        self.form_data["quantity"] = 10
+        self.form_data["status"] = "Accepted"
+
+        response = self.client.post(self.view, data=self.form_data)
+
+        quantity_after = len(
+            Review.objects.filter(
+                decision_sent_date__isnull=True, status="Accepted"
+            ).all()
+        )
+
+        self.assertEqual(len(mail.outbox), 7)
+        self.assertRedirects(response, self.view)
+        self.assertEqual(quantity_before, quantity_after + 7)
+
+    def test_send_all_waitlisted_within_current_date_range(self):
```

Comment:
Is there any reason we'd expect the behaviour for waitlisted status (and rejected below) to differ from the test for accepted above, since they end up just getting passed in to the queryset filter, warranting separate tests? 

If there's something different about these tests that makes them necessary, that would be helpful to call out in a docstring